### ISSUE TITLE
Unexpected Redirects to Login Page in Django Admin

### ISSUE THREAD

**Issue by lukaszsi** [heroku/python-getting-started#228]
I am experiencing an intermittent issue with the Django Admin application while going through this app and Heroku basic tutorial: https://devcenter.heroku.com/articles/getting-started-with-python. All I did are steps in the tutorial, including setting up db extension + the only additional thing is uncommenting admin functionality in urls and settings, that's it.

The issue itself:
When navigating through the admin interface, I am unexpectedly redirected to the login page. This behaviour seems to occur randomly; sometimes I can click through several pages without any issues, but then suddenly I am logged out and asked to log in again.

Moreover, I have noticed that in some instances, using the browser's "Back" button allows me to return to the previous page and I appear to be logged back in, which suggests that I was not actually logged out but rather incorrectly redirected.

Here are some details that might be relevant to this issue:

- **Browser:** I was checking it both in Chrome and Firefox, including "incognito mode"
- **Resources** Testing it on eco and basic dynos
- **Steps to Reproduce:**
  1. Log into the Django Admin deployed to Heroku.
  2. Navigate through various admin pages.
  3. Observe the random redirects to the login page.

**Comment by edmorley** [comment]
@lukaszsi Hi!

The Django admin feature is disabled by default in the app in this repository, since the "Getting Started with Python" tutorial doesn't use/need the admin panel, and it's best to not have the admin panel enabled on all sites that clone this repository as a template (those that do need it can enable it themselves).

As such, anything to do with the admin panel is really out of scope of the guide and as such this app (beyond something in the code in this repo conflicting with someone choosing to enable the admin panel themselves later).

I've just tried out the admin panel myself locally and it worked fine.

The only way I could reproduce your issue was if I restarted the server after having not set an explicit `DJANGO_SECRET_KEY`. Having sessions be invalidated in such a case is expected, and mentioned here:
https://github.com/heroku/python-getting-started/blob/f3145f313df02f8bf5d19b88f09308a015f11dd1/gettingstarted/settings.py#L26-L38

Set an explicit `DJANGO_SECRET_KEY` config var on your app and the sessions will be preserved across server restarts. (The default of the secret being random on boot is to remove one of the pain points of getting started locally.)
